src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -125,6 +125,10 @@ public void updateSessionOnSuccess() {
         sessionWrapper.addExtraToCurrentFrameStep(EXTRA_POST_SUCCESS, true);
     }
 
+    public Object getSessionSuccessStatus() {
```

Comment:
think we should return a boolean here ?

Reply:
I was thinking about that, but that means we would need to also check if `sessionWrapper.getCurrentFrameStepExtra(EXTRA_POST_SUCCESS);` is null and return `false` in that case. Would `false` and `null` always mean the same thing? Or would there be cases where we would want to differentiate the two?

Reply:
yeah think `false` and `null` are same things here as they both signify that no post request has been successful so far.

Reply:
mmm i'm thinking they don't necessarily mean the same thing. Since `null` means no post request has been attempted yet while `false` means a post request has been attempted but not successful. 

I think this distinction matters because for this use case, if it is already set to `false`, then calling `updateSessionOnSuccess` will result in this [error](https://github.com/dimagi/commcare/blob/ad23db87373ae0d65795751cb8610174fd757621/src/main/java/org/commcare/suite/model/StackFrameStep.java#L176-L179). So that will need to be handled differently than `null`

Reply:
I see, that makes sense to me. Maybe we can have this method as below instead to represent more literally what we are using this to check - 

````
public Boolean isPostAlreadyDone() {
return sessionWrapper.getCurrentFrameStepExtra(EXTRA_POST_SUCCESS) == null ;
}

````

Reply:
ah ya i like that suggestion. thank you https://github.com/dimagi/commcare-core/pull/1433/commits/5fa972268075657ba740298b3415c3c67c41c39d